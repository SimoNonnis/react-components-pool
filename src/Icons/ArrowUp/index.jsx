import React from 'react';
import PropTypes from 'prop-types';

/**
 * ArrowUp Icon.
 * @param {number} width
 * @param {number} height
 * @param {string} fill
 * @param {string} className
 * @param {string} classNamePath
 * @returns {React.Component}
 */
function ArrowUp({ width, height, fill, className, classNamePath }) {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 240.835 240.835"
      width={width}
      height={height}
      className={className}
    >
      <title>ArrowUp Icon</title>
      <path
        className={classNamePath}
        fill={fill}
        d="M129.007,57.819c-4.68-4.68-12.499-4.68-17.191,0L3.555,165.803c-4.74,4.74-4.74,12.427,0,17.155 c4.74,4.74,12.439,4.74,17.179,0l99.683-99.406l99.671,99.418c4.752,4.74,12.439,4.74,17.191,0c4.74-4.74,4.74-12.427,0-17.155 L129.007,57.819z"
      />
    </svg>
  );
}

ArrowUp.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  fill: PropTypes.string,
  className: PropTypes.string,
  classNamePath: PropTypes.string,
};

ArrowUp.defaultProps = {
  width: 40,
  height: 40,
  fill: 'currentColor',
  className: 'arrowUpIcon',
  classNamePath: 'arrowUpIconPath',
};

export default ArrowUp;
