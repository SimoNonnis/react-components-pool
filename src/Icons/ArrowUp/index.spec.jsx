/* eslint-env jest */
import React from 'react';

import ArrowUp from './';

describe('Test <ArrowUp /> icon', () => {
  it('should render ArrowUp icon correctly with default props', () => {
    const wrapper = shallow(<ArrowUp />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render ArrowUp icon correctly with props', () => {
    const wrapper = shallow(
      <ArrowUp fill="#a4a1a1" className="testClassname" height={30} width={30} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
