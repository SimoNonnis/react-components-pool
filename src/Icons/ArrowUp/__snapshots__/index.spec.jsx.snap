// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ArrowUp /> icon should render ArrowUp icon correctly with default props 1`] = `
<svg
  className="arrowUpIcon"
  height={40}
  viewBox="0 0 240.835 240.835"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    ArrowUp Icon
  </title>
  <path
    className="arrowUpIconPath"
    d="M129.007,57.819c-4.68-4.68-12.499-4.68-17.191,0L3.555,165.803c-4.74,4.74-4.74,12.427,0,17.155 c4.74,4.74,12.439,4.74,17.179,0l99.683-99.406l99.671,99.418c4.752,4.74,12.439,4.74,17.191,0c4.74-4.74,4.74-12.427,0-17.155 L129.007,57.819z"
    fill="currentColor"
  />
</svg>
`;

exports[`Test <ArrowUp /> icon should render ArrowUp icon correctly with props 1`] = `
<svg
  className="testClassname"
  height={30}
  viewBox="0 0 240.835 240.835"
  width={30}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    ArrowUp Icon
  </title>
  <path
    className="arrowUpIconPath"
    d="M129.007,57.819c-4.68-4.68-12.499-4.68-17.191,0L3.555,165.803c-4.74,4.74-4.74,12.427,0,17.155 c4.74,4.74,12.439,4.74,17.179,0l99.683-99.406l99.671,99.418c4.752,4.74,12.439,4.74,17.191,0c4.74-4.74,4.74-12.427,0-17.155 L129.007,57.819z"
    fill="#a4a1a1"
  />
</svg>
`;
